dids/src/main/kotlin/web5/sdk/dids/did/BearerDid.kt

```diff
@@ -61,6 +62,41 @@ public class BearerDid(
     return Pair(signer, verificationMethod)
   }
 
+  /**
+   * Adds a new service to the DID Document and returns a new `BearerDid` instance with the updated document.
+   *
+   * @param service The service to add to the DID Document.
+   * @return A new `BearerDid` instance with the updated DID Document.
+   */
+  public fun addService(service: Service): BearerDid {
+    val updatedServices = document.service?.toMutableList() ?: mutableListOf()
+    updatedServices.add(service)
+    val updatedDocument = createUpdatedDocument(updatedServices)
+    return BearerDid(uri, did, keyManager, updatedDocument)
+  }
```

Comment:
would it make more sense for `addService` to be a method on `DidDocument`?  that's what we've done in [go](https://github.com/TBD54566975/web5-go/blob/main/dids/didcore/document.go#L195-L198) and [dart](https://github.com/TBD54566975/web5-dart/blob/main/packages/web5/lib/src/dids/did_core/did_document.dart#L172-L175).

this method is helpful for building did docs in general. examples [here](https://github.com/TBD54566975/web5-dart/blob/main/packages/web5/lib/src/dids/did_dht/did_dht.dart#L75-L77) and [here](https://github.com/TBD54566975/web5-dart/blob/main/packages/web5/lib/src/dids/did_web/did_web.dart#L74-L76). same usage exists in go as well

Reply:
that would only work if the service was mutable (like it is in other places) - otherwise currently, has to return a new did (with it added to the doc) OR you need to return a new doc and then update the doc in the did (which returns a new did) if sticking with immutable approach in kotlin (correct me if wrong).

Reply:
ie, you would do blah.document.addService()... but it wouldn't change document in place but return a new document as vaaaaaals: 

![image](https://github.com/TBD54566975/web5-kt/assets/14976/4da09549-77da-4a3e-9620-6708c5e52ae5)

Reply:
so - we could change them to be mutable lists?